runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -61,6 +63,22 @@
 public class CorfuTable<K ,V> implements
         ICorfuTable<K, V>, ICorfuSMR<CorfuTable<K, V>> {
 
+    // Accessor/Mutator threads can interleave in away that creates a deadlock because the can create a
+    // circular dependency between the VersionLockedObject(VLO) lock and the common forkjoin thread pool. In order
+    // to break the dependency, parallel stream operations have to execute on a separate pool that applications
+    // cant use. For example, if there are 4 accessor threads all using the common forkjoin pool, one of the threads
+    // can acquire the VLO lock and cause the other 3 threads to wait, but after acquiring the VLO lock, the thread
+    // gets block on parallel stream, because the pool is exhauseted with threads that are trying to acquire the VLO
+    // look, which creates a circular decadency. In other words, a deadlock.
+    private static final ForkJoinPool.ForkJoinWorkerThreadFactory factory = pool -> {
+        final ForkJoinWorkerThread worker = ForkJoinPool.defaultForkJoinWorkerThreadFactory.newThread(pool);
+        worker.setName("CorfuTable-Forkjoin-pool-" + worker.getPoolIndex());
+        return worker;
+    };
+
+    private final static ForkJoinPool pool = new ForkJoinPool(Runtime.getRuntime().availableProcessors() - 1,
```

Comment:
Should this be a parameter ? or is this formula always good.

Reply:
This maintains the current behavior, which uses the common thread pool. The common thread pool uses # of cpus - 1. 

 I don't like it, but making this configurable is part of a bigger change to be able to pass a client context to different layers.